src/UI/Areas/Identity/Pages/Account/Manage/DeletePersonalData.cshtml.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Identity.UI.Pages.Account.Manage
+{
+    public class DeletePersonalDataModel : PageModel
+    {
+        private readonly UserManager<IdentityUser> _userManager;
+        private readonly SignInManager<IdentityUser> _signInManager;
+        private readonly ILogger<PrivacyModel> _logger;
+
+        public DeletePersonalDataModel(
+            UserManager<IdentityUser> userManager,
+            SignInManager<IdentityUser> signInManager,
+            ILogger<PrivacyModel> logger)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _logger = logger;
+        }
+
+        public async Task<IActionResult> OnGet()
+        {
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+            }
+
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            var user = await _userManager.GetUserAsync(User);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with ID '{_userManager.GetUserId(User)}'.");
+            }
+
+            var result = await _userManager.DeleteAsync(user);
+            if (!result.Succeeded)
+            {
+                throw new ApplicationException($"Unexpected error occurred deleteing user with ID '{user.Id}'.");
+            }
+
+            await _signInManager.SignOutAsync();
+
+            _logger.LogInformation("User with ID '{UserId}' deleted themselves.", _userManager.GetUserId(User));
+
+            // REVIEW: should this redirect to somewhere that says user was deleted?
+            return RedirectToPage("./");
```

Comment:
This should redirect to the website root `return Redirect("~/");` or a successfully deleted page. Redirecting to the page `./` is kinda weird.

Reply:
So most things in manage are doing some form of RedirectToPage...  is one better than the other?

Reply:
RedirectToPage expects to resolve a page. Since there isn't a page named `./`, it ends up resolving the empty string which ends up in the app root. But that's just a weird way of going about it.

Reply:
Yeah so after they get deleted, manage will no longer be valid, so I really was just trying to redirect to home or whatever.  Unless we think its worth adding a dedicated page for a Deleted confirmation